TeamProject/Engine/Engine/TileGrid/TileMap.cs

```diff
@@ -43,9 +48,89 @@ public Tile GetTileAtPosition(Vector2 position)
                y < Tiles.GetLength(1))
             {
                 t = Tiles[x, y];
-            }
+            }            
+
             return t;
         }
 
+        public void Update(GameTime gameTime)
+        {
+            if ((gameTime.TotalGameTime.Milliseconds % 2000 ) == 0)
```

Comment:
Unsure if this line will cause some problems.

A game running at 60 frames a seccond will have a frame time of 16ms. Which means that `TotalGameTime.Milliseconds` will incrament by 16 every `Update()`. It is likely then to miss an exact multiple of 2000, causing the code to not run. You also can't assume that it will be exacly 16ms every frame.

The behavior of this current code then is based on the framerate, and likely will skip more times than not. Instead you should have a `timeUntillTempTick` field variable, decrement it by `GameTime.ElapsedGameTime` and check if it is less than 0. That way the code is guaranteed to run on the next frame, after 2000ms have elapsed.

See `Bush.Update` for how I'm doing it there.
https://github.com/JR-Morgan/TeamProject-Game/blob/26f858fb9fed23620511d1b3c42c5b85b918ba8d/TeamProject/GlobalWarmingGame/Interactions/Interactables/Bush.cs#L59-L71